pkg/common/resources/resources.go

```diff
@@ -455,6 +455,74 @@ func (r *Resource) fitIn(smaller *Resource, skipUndef bool) bool {
 	return true
 }
 
+// Denominator Resources can be either guaranteed Resources or fairmax Resources.
+// If the quanity is explicitly 0 or negative, we will check usage.  If usage >= 0, the share will be set to 1.0.  Otherwise, it will be set 0.0.
+func getShareFairForDenominator(resourceType string, allocated Quantity, denominatorResources *Resource) (float64, bool) {
+	if denominatorResources == nil {
+		return 0.0, false
+	}
+
+	denominator, ok := denominatorResources.Resources[resourceType]
+
+	switch {
+	case ok && denominator <= 0:
+		if denominator < 0 {
+			log.Log(log.Resources).Debug("denominator is negative. will not compute share",
+				zap.String("resource key", resourceType),
+				zap.Int64("resource quantity", int64(denominator)))
+		}
+
+		if allocated <= 0 {
+			//explicit 0 or negative value with NO usage
+			return float64(0.0), true
```

Comment:
I don't believe the cast is necessary here.

Reply:
done